CHANGELOG.md

```diff
@@ -6,6 +6,15 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+
+### Fixed
+
+- LSP: `renderDiagnostic` and `explainError` skipped diagnostics if they were in the same location
+  as other diagnostics
+- LSP: `renderDiagnostic` and `explainError` stopped searching early and defaulted to the first
+  diagnostic in the file instead of the next diagnostic after the current cursor position
```

Comment:
```suggestion
- LSP: `renderDiagnostic` and `explainError` skipped diagnostics
  if they were in the same location
  as other diagnostics
- LSP: `renderDiagnostic` and `explainError` stopped searching early
  and defaulted to the first
  diagnostic in the file,
  instead of the next diagnostic after the current cursor position
```